src/index.ts

```diff
@@ -12,11 +12,7 @@ declare global {
   interface Window {
     CoinbaseWalletSDK: typeof CoinbaseWalletSDK;
     CoinbaseWalletProvider: typeof CoinbaseWalletProvider;
-    /**
-     * Workaround possible compatibility issues in Angular
-     * Set as CoinbaseWalletProvider
-     */
-    ethereum?: any;
+    ethereum?: unknown;
```

Comment:
i wonder if we could instead declare `unknown | PreciseType` so that type hinting is more useful.

Reply:
I had first attempted that, but it doesn't seem to get picked up. 
> // In a union an unknown absorbs everything

from https://github.com/microsoft/TypeScript/pull/24439

![image](https://user-images.githubusercontent.com/50590950/160176928-dc504548-3b62-47eb-be58-7559e38cc3b4.png)

and `coinbaseWalletExtension`:
![image](https://user-images.githubusercontent.com/50590950/160177010-c2b15431-ea84-430c-ad2c-d31ac548cc4d.png)

I'll add back a comment

Reply:
ah ok, oof then.

Reply:
@ljharb Thoughts on addressing this issue? https://github.com/coinbase/coinbase-wallet-sdk/issues/343

> All declarations of 'ethereum' must have identical modifiers

In order to fix the above error, the type would need to be updated to:

```ts
interface Window {
  ethereum: any | undefined;
}
```

But I hesitate to make changes based on a conflicting library's type defs